Java/core/src/main/java/com/amazon/randomcutforest/store/PointStoreFloat.java

```diff
@@ -135,7 +135,7 @@ public String toString(int index) {
     }
 
     public float[] getStore() {
-        return store;
+        return Arrays.copyOf(store, getValidPrefix() * dimensions);
```

Comment:
I think it would be better to perform the truncation in the mapper class:
* most of the time, people don't expect business logic in getters
* since the logic to inflate the array appears in the mapper, it makes sense for the logic to truncate the array to be there too.

Reply:
Done